doc/charm++/manual.rst

```diff
@@ -5796,19 +5796,21 @@ is created, a *reference* is returned immediately. However, if the
 *value* calculated by the future is needed, the calling program blocks
 until the value is available.
 
-Charm++ provides all the necessary infrastructure to use futures by
-means of the following functions:
+We provide both C-compatible and object-oriented interfaces for using
+futures, which include the following functions:
 
 .. code-block:: c++
 
-    CkFuture CkCreateFuture(void)
-    void CkReleaseFuture(CkFuture fut)
-    int CkProbeFuture(CkFuture fut)
-    void *CkWaitFuture(CkFuture fut)
-    void CkSendToFuture(CkFuture fut, void *msg)
+   CkFuture CkCreateFuture(void); /* or */ ck::future<T>()
+   void CkReleaseFuture(CkFuture fut); /* or */ void ck::future<T>::release()
+   int CkProbeFuture(CkFuture fut); /* or */ bool ck::future<T>::probe()
+   void *CkWaitFuture(CkFuture fut); /* or */ T ck::future<T>::get()
+   void CkSendToFuture(CkFuture fut, void *msg); /* or */ void ck::future<T>::set(T)
```

Comment:
It might be more readable to either make these two separate lists, or to make a table with a C column and a C++ column.